lib/components/CdeModal.tsx

```diff
@@ -19,23 +29,29 @@ const CdeModal: FC = () => {
         }
     }, [errorMessage, setErrorMessage]);
 
-    const renderStepComponent = () => {
+    const renderStepComponent = (): ReactElement => {
         switch (step) {
             case STEPS.HOME:
                 return <Home/>;
-            case STEPS.REPOSITORY:
+            case STEPS.COLLECTION:
                 return <MappingStep/>;
-            // Add cases for other steps
+            // Add cases for other steps with commonProps
             default:
                 return <div>Unknown step</div>;
         }
     };
 
-
     return (
         <>
-            <Modal open={isOpen} onClose={handleClose} maxWidth="xl">
-                {renderStepComponent()}
+            <Modal open={isModalOpen} onClose={onClose} maxWidth="xl" isInfoOpen={isInfoOpen}>
+                {loadingMessage ? <Loading loadingMessage={loadingMessage}/> :
+                    <>
+                        <Header onClose={onClose} isInfoOpen={isInfoOpen} setIsInfoOpen={setIsInfoOpen}/>
```

Comment:
I moved this out of the ModalLayout because it seems to me that we want all the Steps to have the Header. If that's not the case then we can move it to the renderStepComponent